controllers/chat/getSignedChannelList.js

```diff
@@ -62,7 +82,7 @@ const getSignedChannelList = async (req, res) => {
       queriedChannels.push(...response);
     });
 
-    const channels = queriedChannels.map(__transformChannelData);
+    const channels = queriedChannels.reduce(__filterAndTransformChannelData, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `reduce` here is more efficient than the previous `map` operation because it avoids creating an intermediate array. However, it's important to note that the behavior has changed: previously, all channels were transformed, but now only some are (those that pass the filter). Make sure this change is intentional and won't break anything.


<!-- This is an auto-generated comment by OSS CodeRabbit -->